### ISSUE TITLE
Bad look and feel around "minishift config view" output

### ISSUE THREAD

**Issue by amitkrout** [minishift/minishift#724]
This is more towards a cosmetic issue around "minishift config view". 
```
Environment : 
minishift v1.0.0-rc.1
```

```
$minishift config set cpus 4
$minishift config set memory 2096
$ minishift config view
- cpus                 : 4
- memory               : 2096$
```
The output would be better if it looks like this 
```
$ minishift config view
- cpus                 : 4
- memory               : 2096
$
```

**Comment by hferentschik** [comment]
So to be clear, what you are seeing is that you don't get a newline after the last line of the config. Instead you get a prompt right after '- memory               : 2096'. If this is the case I would consider it a bug, but I atm cannot re-produce it. For me the prompt appears as expected on a newline. Also looking at the source https://github.com/minishift/minishift/blob/4d6d0d30e8f37982e32c1b2b849ad5d17954e917/cmd/minishift/cmd/config/view.go#L95-L95, a newline is explicitly added.

Is this reproducible behaviour for you? If so on which OS and which shell type are you using?

**Comment by amitkrout** [comment]
@hferentschik confirming that the prompt doesn't appear on a new line on mac OS Sierra ```/bin/bash``` terminal and even on fedora 23 ```/bin/bash``` terminal

I checked the same on windows 10 powershell where the prompt appears on a new line as expected 
```
PS C:\v1.0.0-rc-1> .\minishift.exe config view
PS C:\v1.0.0-rc-1> .\minishift.exe config set cpus 4
PS C:\v1.0.0-rc-1> .\minishift.exe config set memory 2096
PS C:\v1.0.0-rc-1> .\minishift.exe config view
- cpus                 : 4
- memory               : 2096
PS C:\v1.0.0-rc-1> 
```

**Comment by hferentschik** [comment]
@LalatenduMohanty, @praveenkumar, @budhrg - can you guys please double check on your machines.

**Comment by praveenkumar** [comment]
@hferentschik confirm actually I wanted to file it but looks like @amitkrout was fast :), I am going to work on that and send PR.

**Comment by hferentschik** [comment]
+1, thanks. But also keep the priorities in mind ;-)

**Comment by praveenkumar** [comment]
@hferentschik  Ack 👍

**Comment by agajdosi** [comment]
@hferentschik @praveenkumar I am facing problems with `--format` flag of `minishift config view` command. For integration tests I would like to verify that expected values are in stdout of minishift config view. However extra blank spaces are not so friendly, so I tried to use --format flag.

I would expect that using default string mentioned in help of command 
`"- {{.ConfigKey | printf "%-21s"}}: {{.ConfigValue}"` will print same output as without flag. But it throws an error:  ```E0410 17:44:48.087952   24007 view.go:80] Error creating view template: template: view:1: unexpected "%" in operand``` Seems like problem with quotes.

Tried it with combination of " and ', but no luck. Also tried to use native value directly taken from view.go: `"- {{.ConfigKey | printf \"%-21s\"}}: {{.ConfigValue}}\n"`. Also nothing.

What I have find as a workaround, is to define the format string on two lines:
```
[agajdosi@dhcp-24-156 ~]$ minishift config view --format "-{{.ConfigKey}}:{{.ConfigValue}}
> "
-disk-size:25g
-memory:3333[agajdosi@dhcp-24-156 ~]$ 
```
However this is still ugly. Is it expected to be like this, am I overlooking something or is it behaving badly?

**Comment by hferentschik** [comment]
@agajdosi, I think you are onto something here. I think this is all related. To start with the pattern should not include the '\n'. This is combination with the sorting of the config options is causing the issue. I think we need to fix on two ends. First on the default template side, then at the iteration of the key/values and maybe finally even at the sorting side.

**Comment by hferentschik** [comment]
@agajdosi, regarding getting "extended" formatting into the template, this will be tricky, since you are running into escaping issue again. Using the "simple" format (just the struct names should work). This I consider a but.

**Comment by hferentschik** [comment]
Resolved via pull request #730
